resources/views/sports/create.blade.php

```diff
@@ -3,71 +3,122 @@
 @section('content')
     <div class="container">
         <div class="row justify-content-center">
-            <div class="col-md-8">
-                <form method="POST" action="{{ route('store') }}">
+            <div class="col-md-8" x-data="{
+                onFormSubmit(event){
+                    const form = event.target
+                    const spans = form.querySelectorAll('span.invalid-feedback')
+                    if(spans.length == 0){
+                        form.submit()
+                    }else {
+                        // focus in input element
+                        spans[0].previousElementSibling.focus()
+                    }
+                }
+            }">
+
+                @if ($errors->any())
+                    <div class="alert alert-danger text-center">
+                        Invalid information
+                    </div>
+                @endif
+                <form method="POST" action="{{ route('store') }}" @submit.prevent="onFormSubmit($event)">
                     @foreach ($sports as $sport)
-                        <div class="card mb-4">
+                        <div class="card mb-4" x-data="{
```

Comment:
I never uses Alpine JS, but Js code in the middle of html, it's a normal practice?)

Reply:
Yes, you can use in the middle of HTML or a small portion of section. You can also use in the bottom script.

Reply:
With alpine especially it is. But personally I would use it only if it's very short

Reply:
> Yes, you can use in the middle of HTML or a small portion of section. You can also use in the bottom script.

While you CAN, in this case I don't think you SHOULD. It's cool if it's a few lines of code, but now you've put the whole "mini-application" inside the tag. Hard to read and to separate HTML from JS. I would define functions at the bottom somewhere.

Reply:
Yes, I agree with you @PovilasKorop . First time I just used backend validation and send pull request. After a sudden, I thought why not use a new way to validate form. Then I implement Alpine JS. Although I never use alpine js for form validation. It was an experimental implementation and I make it.